src/app/filelogger.cpp

```diff
@@ -87,32 +80,41 @@ void FileLogger::configure()
         m_logFile = new QFile(m_path);
         openLogFile();
     }
+}
 
-    m_backup = pref->fileLogBackup();
-    m_size = pref->fileLogMaxSize();
+void FileLogger::deleteOld(const int age, const FileLogAgeType ageType)
+{
+    QDateTime date = QDateTime::currentDateTime();
+    QDir dir(m_path);
 
-    if (pref->fileLogDeleteOld()) {
-        QDateTime date = QDateTime::currentDateTime();
+    switch (ageType) {
+    case DAYS:
+        date = date.addDays(age);
+        break;
+    case MONTHS:
+        date = date.addMonths(age);
+        break;
+    default:
+        date = date.addYears(age);
+    }
 
-        switch (static_cast<FileLogAgeType>(pref->fileLogAgeType())) {
-        case DAYS:
-            date = date.addDays(pref->fileLogAge());
+    foreach (const QFileInfo file, dir.entryInfoList(QStringList() << "qbittorrent.log.bak*", QDir::Files | QDir::Writable, QDir::Time | QDir::Reversed)) {
```

Comment:
Can you try `{ "qbittorrent.log.bak*" }` instead of `QStringList() << "qbittorrent.log.bak*"`?